### PR TITLE
Improve meeting with multiple Group and Space messages

### PR DESCRIPTION
**Pull request by gparant**
**First version of the meeting moderation:**
Group
- [x] kick message ;
- [x] mute video message ;
- [x] mute message video to every attendees ;
- [x] mute microphone message ;
- [x] mute message microphone to every attendees ;

Space
- [x] kick message by attendee ;
- [x] mute video message ;
- [x] mute message video to every attendees ;
- [x] mute microphone message ;
- [x] mute message microphone to every attendees ;

Migrate block and report
- [x] Add block user in the action menu
- [x] Add report user in the action menu

Test E2E
- [x] kick message by attendee ;
- [x] mute video message ;
- [x] mute message video to every attendees ;
- [x] mute microphone message ;
- [x] mute message microphone to every attendees ;

**second version:**
Merge request created here: #3671

### CODE REVIEW

File: messages/protos/messages.proto [hunk 1]
```diff
@@ -622,6 +637,35 @@ message MegaphoneEnabledMessage {
...
+}
+
+message MutedMessage {
+  string userUuid = 1;
+  string message = 2;
+}
+
+message MuteVideoParticipantIdSpaceMessage {
+  string spaceName = 1;
+  string value = 2;
```
Comment (moufmouf) [inline messages/protos/messages.proto:657]: Rename `value` to `mutedUserUuid`

File: messages/protos/messages.proto [hunk 2]
```diff
@@ -622,6 +637,35 @@ message MegaphoneEnabledMessage {
   bool value = 1;
 }
 
+message MuteParticipantIdSpaceMessage {
+  string spaceName = 1;
+  string value = 2;
```
Comment (moufmouf) [inline messages/protos/messages.proto:642]: Rename `value` to `mutedUserUuid`

File: messages/protos/messages.proto [hunk 3]
```diff
@@ -622,6 +637,35 @@ message MegaphoneEnabledMessage {
...
+}
+
+message MuteEveryBodySpaceMessage {
+  string spaceName = 1;
+  string userId = 2;
```
Comment (moufmouf) [inline messages/protos/messages.proto:647]: Maybe rename to `notMutedUserUuid` for additional clarity?

Let's ALWAYS use "uuid" if this is a string, and "id" if this is the number.

File: back/src/Services/SocketManager.ts [hunk 4]
```diff
@@ -1500,6 +1589,106 @@ export class SocketManager {
...
+                        userUuid: user.uuid,
+                        message: "muted",
+                    },
+                },
+            };
+            userMuted.socket.write(serverToClientMessage);
+        }
+    }
+
+    handlerMuteEveryBodySpaceMessage(user: User) {
```
Comment (moufmouf) [inline back/src/Services/SocketManager.ts:1628]: Check all the names of the methods.

If should start with `handle` (the verb). Not `handler` or `hander`

File: back/src/Services/SocketManager.ts [hunk 5]
```diff
@@ -1500,6 +1589,106 @@ export class SocketManager {
...
+        const group = user.group;
+        if (!group) {
+            return;
+        }
+        for (const userMuted of group.getUsers().values()) {
```
Comment (moufmouf) [inline back/src/Services/SocketManager.ts:1633]: That would be `mutedUser` instead of `userMuted`. In English, the adjective goes before the noun.

File: messages/protos/messages.proto [hunk 6]
```diff
@@ -755,6 +801,11 @@ message PusherToBackMessage {
     AskPositionMessage askPositionMessage = 22;
     EditMapCommandMessage editMapCommandMessage = 23;
     PingMessage pingMessage = 24;
+    KickOffUserMessage kickOffUserMessage = 25;
+    MuteParticipantIdSpaceMessage muteParticipantIdSpaceMessage = 26;
+    MuteEveryBodySpaceMessage muteEveryBodySpaceMessage = 27;
+    MuteVideoParticipantIdSpaceMessage muteVideoParticipantIdSpaceMessage = 28;
+    MuteVideoEveryBodySpaceMessage muteVideoEveryBodySpaceMessage = 29;
```
Comment (moufmouf) [inline messages/protos/messages.proto:808]: All those messages are called "SpaceMessage" but they do not apply to spaces, right?
You should rename those (remove the "space" word everywhere when it is a direct message that applies to groups)

File: back/src/Services/SocketManager.ts [hunk 7]
```diff
@@ -1461,6 +1466,90 @@ export class SocketManager {
...
+                kickOffSpaceUserMessage: {
+                    spaceName: kickUserMessage.spaceName,
+                    userId: kickUserMessage.userId,
+                    filterName: kickUserMessage.filterName,
+                },
+            },
+        });
+    }
+
+    handleMuteMicrophoneSpaceUserMessage(
```
Comment (moufmouf) [inline back/src/Services/SocketManager.ts:1484]: All the methods below do not apply to "spaces". They target groups.
So there should be no "space" in the method names of this class, right?

File: libs/map-editor/src/types.ts [hunk 8]
```diff
@@ -335,4 +335,5 @@ export enum GameMapProperties {
     WRITABLE_BY = "writableBy",
     ZONE = "zone",
     ZOOM_MARGIN = "zoomMargin",
+    TEAMS_ROOM = "teamsRoom",
```
Comment (moufmouf) [inline libs/map-editor/src/types.ts:338]: This probably should be in another PR as this PR is unrelated to Teams.

File: play/src/front/Connection/RoomConnection.ts [hunk 9]
```diff
@@ -409,6 +413,60 @@ export class RoomConnection implements RoomConnection {
...
+                                // If the sender receive this message, ignore it
+                                if (subMessage.muteVideoEverybodySpaceUserMessage.userId === this.userId?.toString())
+                                    break;
+                                notificationPlayingStore.playNotification(
+                                    get(LL).notification.askToMuteCamera(),
+                                    "camera-off.png"
+                                );
+                                requestedCameraState.disableWebcam();
+                                break;
+                            }
```
Comment (moufmouf) [inline play/src/front/Connection/RoomConnection.ts:469]: Can we apply inversion of control here?
I'm trying to keep the `RoomConnection` ignorant of the rest of the code.
I would need to keep it this way to restore the "benchmark" feature some day.

Ideally, the benchmark feature could use the same "RoomConnection" class as the rest of the code. For this to work, the class must not depend on anything that is related to the front (no svelte store, no direct reference to a service, etc...)
Instead, it should just emit messages that are listened out of the class.

This code should therefore be migrated out of the RoomConnection, inside the GameScene or another dedicated class.

File: play/src/front/Connection/RoomConnection.ts [hunk 10]
```diff
@@ -658,6 +716,19 @@ export class RoomConnection implements RoomConnection {
...
+                        "audio-mute.svg"
+                    );
+                    requestedMicrophoneState.disableMicrophone();
+                    break;
+                }
+                case "mutedVideoMessage": {
+                    notificationPlayingStore.playNotification(get(LL).notification.askToMuteCamera(), "camera-off.png");
+                    requestedCameraState.disableWebcam();
+                    break;
+                }
```
Comment (moufmouf) [inline play/src/front/Connection/RoomConnection.ts:731]: Same comment as above. Apply inversion of control, move specific code out of this class.

File: play/src/front/Stores/NotificationStore.ts [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+export type Notification = {
+    id: string;
+    text: string;
+    icon: string;
+};
+
+/**
+ * A store that contains the URL of the sound currently playing
+ */
```
Comment (moufmouf) [inline play/src/front/Stores/NotificationStore.ts:11]: Comment is probably wrong.

File: play/src/front/Streaming/Contract/TackStreamWrapperInterface.ts [hunk 12]
```diff
@@ -0,0 +1,34 @@
+import { Readable } from "svelte/store";
+import { TrackInterface } from "./TrackInterface";
+
+export interface TackStreamWrapperInterface {
```
Comment (moufmouf) [inline play/src/front/Streaming/Contract/TackStreamWrapperInterface.ts:4]:
```suggestion
export interface TrackStreamWrapperInterface {
```

File: play/src/i18n/es-ES/camera.ts [hunk 13]
```diff
@@ -36,6 +36,18 @@ const camera: DeepPartial<Translation["camera"]> = {
...
+        closeMenu: "Cerrar el menú",
+        senPrivateMessage: "Enviar un mensaje privado (próximamente)",
+        kickoffUser: "Expulsar usuario",
+        muteAudioUser: "Silenciar audio",
+        muteAudioEveryBody: "Silenciar audio para todos",
+        muteVideoUser: "Silenciar vídeo",
+        muteVideoEveryBody: "Silenciar vídeo para todos",
+        pin: "Fijar",
+        blockOrReportUser: "Bloquear o informar de usuario",
+    },
```
Comment (moufmouf) [inline play/src/i18n/es-ES/camera.ts:50]: I usually never translate the text of the languages I don't speak.
I have had several feedbacks of German speakers telling us that our translations were wrong. That's because Google Translate cannot translate propertly on such a small context.

I think we should not translate those. The framework will fallback to English, and the community can provide adequate translations. They will be better at this than us.

File: messages/protos/messages.proto (hunk 3 above)
Comment (gparant) [reply messages/protos/messages.proto:647]: or senderUserId maybe?

File: back/src/Services/SocketManager.ts (hunk 7 above)
Comment (gparant) [reply back/src/Services/SocketManager.ts:1484]: In this case is a SpacesWatcher passed in parameter, so we use space.

File: play/src/i18n/es-ES/camera.ts (hunk 13 above)
Comment (gparant) [reply play/src/i18n/es-ES/camera.ts:50]: I use a specific IA to translate the text. I think that we should trust it 💪
